uast/nodes/node.go

```diff
@@ -9,31 +9,41 @@ import (
 const applySort = false
 
 // Equal compares two subtrees.
-func Equal(n1, n2 Node) bool {
+// Equality is checked by value (deep), not by reference.
+func Equal(n1, n2 External) bool {
 	if n1 == nil && n2 == nil {
 		return true
-	} else if n1 != nil && n2 != nil {
-		return n1.Equal(n2)
+	} else if n1 == nil || n2 == nil {
+		return false
+	}
+	if n, ok := n1.(Node); ok {
+		return n.Equal(n2)
+	} else if n, ok = n2.(Node); ok {
+		return n.Equal(n1)
 	}
-	return false
+	return equalExt(n1, n2)
 }
 
-// Node is a generic interface for a tree structure.
+// Node is a generic node interface.
```

Comment:
I think the previous definition was better since it doesn't use the same term it is defining.